### PR TITLE
 SCALRCORE-15443: IaCP Provider > Environment resource

### PR DESCRIPTION
**Pull request by vmotso**
### Changelog
* [x] I have updated CHANGELOG.md

### Documentation
* [x] I have updated resource/data source documentation in [website/docs](https://github.com/Scalr/terraform-provider-scalr/tree/develop/website/docs)

### State
* [ ] My changes affect the state
* [ ] I have included a state migration and a unit test for it

### CODE REVIEW

**Comment by vmotso** [comment]
Acceptance tests are passed:
https://drone.scalr-labs.net/Scalr/int-terraform-provider-scalr/110

File: scalr/resource_scalr_environment.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	scalr "github.com/scalr/go-scalr"
+)
+
+func resourceScalrEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalrEnvironmnetCreate,
```
Comment (lbn) [inline scalr/resource_scalr_environment.go:13]: Typo

File: scalr/resource_scalr_environment.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+				Type:     schema.TypeList,
+				Computed: true,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceScalrEnvironmnetCreate(d *schema.ResourceData, meta interface{}) error {
```
Comment (lbn) [inline scalr/resource_scalr_environment.go:76]: Environmnet -> Environment everywhere please.

**Review by lbn — CHANGES_REQUESTED** [review]

**Comment by vmotso** [comment]
Tests are passed after changes:
https://drone.scalr-labs.net/Scalr/int-terraform-provider-scalr/112

**Review by lbn — APPROVED** [review]

**Review by lbn — APPROVED** [review]

File: scalr/data_source_environment.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+func dataSourceEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	scalrClient := meta.(*scalr.Client)
+
+	envID := d.Get("id").(string)
+
+	log.Printf("[DEBUG] Read configuration of environment: %s", envID)
+
+	environment, err := scalrClient.Environments.Read(ctx, envID)
+	if err != nil {
```
Comment (lbn) [inline scalr/data_source_environment.go:75]: Need to return an error if it's not scalr.ErrResourceNotFound just like in workspaces (SCALRCORE-17490):
https://github.com/Scalr/terraform-provider-scalr/blob/develop/scalr/data_source_workspace.go#L105-L111

**Review by lbn — CHANGES_REQUESTED** [review]

File: scalr/data_source_environment.go (hunk 3 above)
Comment (vmotso) [reply scalr/data_source_environment.go:76]: fixed

**Pull request merged**